### PR TITLE
[shopsys] upgraded doctrine/orm to latest version

### PR DESCRIPTION
**Pull request by TomasLudvik**
| Q             | A
| ------------- | ---
|Description, the reason for the PR| There were some problems with the order of Order Items when upgrading to the newest doctrine/orm. This has been fixed and the restriction to older doctrine/orm has been removed.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://tl-update-doctrine-13.odin.shopsys.cloud
  - https://cz.tl-update-doctrine-13.odin.shopsys.cloud

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -197,7 +197,6 @@
     },
     "conflict": {
         "symfony/symfony": "*",
-        "doctrine/orm": "2.12.0 || >=2.16.0",
```
Comment (grossmannmartin) [inline composer.json:200]: same conflict should be removed in the `packages/framework/composer.json`

File: project-base/app/src/Model/Order/OrderFacade.php [hunk 2]
```diff
@@ -312,6 +312,9 @@ protected function fillOrderProducts(BaseOrder $order, BaseOrderPreview $orderPr
                 $product,
             );
 
+            $this->em->persist($orderItem);
+            $this->em->flush();
```
Comment (grossmannmartin) [inline project-base/app/src/Model/Order/OrderFacade.php:316]: Shouldn't be the fix also made in the `\Shopsys\FrameworkBundle\Model\Order\OrderFacade::fillOrderProducts`?

**Review by grossmannmartin — CHANGES_REQUESTED** [review]

File: project-base/app/src/Model/Order/OrderFacade.php (hunk 2 above)
Comment (TomasLudvik) [reply project-base/app/src/Model/Order/OrderFacade.php:316]: Well it is never called, but I have added it

File: composer.json (hunk 1 above)
Comment (TomasLudvik) [reply composer.json:200]: fixed

**Review by grossmannmartin — APPROVED** [review]

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**